src/xtgeo/grid3d/grid_properties.py

```diff
@@ -1,24 +1,106 @@
 # -*- coding: utf-8 -*-
 
 """Module for Grid Properties."""
-
-
 import hashlib
 import warnings
-from typing import Optional
+from typing import List, Optional
+
+import deprecation
+import numpy as np
+import pandas as pd
 
 import xtgeo
 from xtgeo.common import XTGDescription, XTGeoDialog
 
 from . import _grid3d_utils as utils
-from . import _grid_etc1, _gridprops_etc, _gridprops_import_eclrun
+from . import _grid_etc1, _gridprops_import_eclrun
 from ._grid3d import _Grid3D
 from .grid_property import GridProperty
 
 xtg = XTGeoDialog()
 logger = xtg.functionlogger(__name__)
 
 
+def gridproperties_from_file(
+    pfile,
+    grid,
+    fformat=None,
+    names=None,
+    dates=None,
+    namestyle=0,
+    strict=(True, False),
+):
+    """Import grid properties from file.
+
+    In case of names='all' then all vectors which have a valid length
+    (number of total or active cells in the grid) will be read
+
+    Args:
+        pfile (str or Path): Name of file with properties
+        fformat (str): roff/init/unrst
+        names: list of property names, e.g. ['PORO', 'PERMX'] or 'all'
+        dates: list of dates on YYYYMMDD format, for restart files, or 'all'
+        grid (obj): The grid geometry object (optional if ROFF)
+        namestyle (int): 0 (default) for style SWAT_20110223,
+            1 for SWAT--2011_02_23 (applies to restart only)
+        strict (tuple of (bool, bool)): If (True, False) (default) then an
+            Error is raised if keyword name is not found, or a key-date combination
+            is not found. However, the dates will processed so that non-valid dates
+            are skipped (still, invalid key-date combinations may occur!).
+            If (True, True) all keywords and dates are tried, while (False, False)
+            means that that only valid entries are imported, more or less silently.
+            Saturations keywords SWAT/SOIL/SGAS are not evaluated as they may be
+            derived.
+    Example::
+        >>> grd = xtgeo.grid_from_file(reek_dir + "/REEK.EGRID", fformat='egrid')
+        >>> gps = xtgeo.gridproperties_from_file(
+        ...     reek_dir + "/REEK.INIT",
+        ...     fformat='init',
+        ...     names=["PORO", "PERMX"],
+        ...     grid=grd,
+        ... )
+    """
+    pfile = xtgeo._XTGeoFile(pfile, mode="rb")
+
+    pfile.check_file(raiseerror=ValueError)
```

Comment:
Perhaps add a message to the error?

Reply:
The message will be set by the `check_file` function to `"File {} does not exist or cannot be accessed".format(self.name)`. I think that would suffice here.